src/parser.rs

```diff
@@ -197,7 +197,7 @@ impl Capturer {
     }
 }
 
-pub(crate) struct Parser<'parser> {
+pub struct Parser<'parser> {
```

Comment:
Can you rename the first argument to `Parser::new()` from `remaining` to `buffer`? Just the argument. The struct field is still `remaining`

I assume you need the constructor to be pub?

It's ok if not I think, but the name is not good as public facing.